### PR TITLE
Plugins: Convert BarGaugePanel to RTL

### LINKED ISSUE #48254: Convert `public/app/plugins/panel/bargauge/BarGaugePanel.test.tsx` to RTL

### PR DESCRIPTION
**Pull request by lpskdl**
**What this PR does / why we need it**:
- convert enzyme to RTL
- added new assertions

**Which issue(s) this PR fixes**:

Fixes #48254

### CODE REVIEW

File: public/app/plugins/panel/bargauge/BarGaugePanel.test.tsx [hunk 1]
```diff
@@ -1,106 +1,134 @@
...
+  describe('when there is no data', () => {
+    it('show a "No Data" message', () => {
+      const panelData = buildPanelData();
+
+      render(<BarGaugePanel {...panelData} />);
 
-    it('should render with title "No data"', () => {
-      const displayValue = wrapper.find(`div[data-testid="${valueSelector}"]`).text();
-      expect(displayValue).toBe('No data');
+      expect(screen.getByText(/no data/i)).toHaveTextContent('No data');
```
Comment (Clarity-89) [inline public/app/plugins/panel/bargauge/BarGaugePanel.test.tsx:20]:
```suggestion
      expect(screen.getByText('No data')).toBeInTheDocument();
```
Quite strange to use the same query to match and element and test for it. In this case it's better to test for presence in the DOM.

File: public/app/plugins/panel/bargauge/BarGaugePanel.test.tsx [hunk 2]
```diff
@@ -1,106 +1,134 @@
...
-    });
+          timeRange: createTimeRange(),
+          state: LoadingState.Done,
+        },
+      });
+
+      const { rerender } = render(<BarGaugePanel {...panelData} />);
+      expect(screen.queryByText(/100/)).toBeInTheDocument();
+      expect(screen.queryByText(/firstbarpanel/i)).not.toBeInTheDocument();
+      expect(screen.queryAllByTestId(valueSelector).length).toEqual(1);
```
Comment (Clarity-89) [inline public/app/plugins/panel/bargauge/BarGaugePanel.test.tsx:47]:
```suggestion
      expect(screen.getAllByTestId(valueSelector)).toHaveLength(1);
```
Although, if we only expect one element, `expect(screen.getByTestId(valueSelector)).toBeInTheDocument();` could be better here.

**Review by Clarity-89 — COMMENTED** [review]
Great work 🎉  Left a few minor suggestions to consider.

**Comment by lpskdl** [comment]
thank you @Clarity-89 for the review. All the comments were addressed. Let me know if I need to change anything else.

**Review by Clarity-89 — APPROVED** [review]
Looks good 🚀

**Pull request merged**